libraries/AP_ESC_Telem/AP_ESC_Telem.cpp

```diff
@@ -309,7 +309,10 @@ void AP_ESC_Telem::send_esc_telemetry_mavlink(uint8_t mav_chan)
 
     // loop through groups of 4 ESCs
     const uint8_t esc_offset = constrain_int16(mavlink_offset, 0, ESC_TELEM_MAX_ESCS-1);
-    const uint8_t num_idx = ESC_TELEM_MAX_ESCS/4;
+    
+    //This ensures that number of escs are present in a group of 4 and if not then it will rectify it by adding 1
+    const uint8_t num_idx = (ESC_TELEM_MAX_ESCS % 4 == 0) ? ESC_TELEM_MAX_ESCS / 4 : (ESC_TELEM_MAX_ESCS / 4) + 1; 
```

Comment:
```suggestion
    // ensure we send out partially-full groups:
    const uint8_t num_idx = (ESC_TELEM_MAX_ESCS + 3) % 4;
```

Reply:
Peter (ESC_TELEM_MAX_ESCS + 3) % 4 this logic would increase the number of loops and also will behave abruptly 
eg. Truth Table
ESC_TELEM_MAX_ESCS -> num_idx -> no. of loops by  logic-> Actual it should be
4->3->3->1
6->1->1->2
1->0->0->1
5->0->0->2
9->0->0->4
As we can see it will also leave some loops/escs as in cases eg. 6 loop should run twice but will run only once by this logic.
Also we can see in case of 1,5,9,... and so on it will fail as it will run 0 time or can say it will skip the loop 
So I think this logic seems to fail.

Proposed Solution Truth Table
(ESC_TELEM_MAX_ESCS % 4 == 0) ? ESC_TELEM_MAX_ESCS / 4 : (ESC_TELEM_MAX_ESCS / 4) + 1
Define: This solution states that if escs are divisible by 4 that means they are already in group of 4 and if not then it can be covered by adding a 1 or can say 1 more loop to cover them.
Truth Table
ESC_TELEM_MAX_ESCS -> num_idx -> no. of loops by  logic-> Actual it should be
4->1->1->1
6->2->2->2
1->1->1->1
5->2->2->2
13->4->4->4

please review it again @peterbarker

Reply:
> I have updated the problems arising in the very first statement(pull req statement) please take look at it once again and solution/logic (ESC_TELEM_MAX_ESCS + 3) % 4 will give absurd result for many values and will throw error for 1,5,9,...

Reply:
> 

I have updated the list of problems arising in the very first statement(pull req statement) please take look at it once again and solution/logic (ESC_TELEM_MAX_ESCS + 3) % 4 will give absurd result for many values and will throw error for 1,5,9,...

Reply:
This solution seems to have some issues correct me if I am wrong
(ESC_TELEM_MAX_ESCS + 3) % 4 this logic would increase,decrease or skip the loops, it will behave abruptly
eg.
Truth Table
ESC_TELEM_MAX_ESCS -> num_idx -> no. of loops by logic-> Actual it should be
4->3->3->1
6->1->1->2
1->0->0->1
5->0->0->2
9->0->0->4
As we can see it will also leave some loops/escs as in cases eg. 6 loop should run twice but will run only once by this logic.
Also we can see in case of 1,5,9,... and so on it will fail as it will run 0 time or can say it will skip the loop
So I think this logic seems to fail.

Proposed Solution Truth Table
(ESC_TELEM_MAX_ESCS % 4 == 0) ? ESC_TELEM_MAX_ESCS / 4 : (ESC_TELEM_MAX_ESCS / 4) + 1
Define: This solution states that if escs are divisible by 4 that means they are already in group of 4 and if not it means it is partially filled then it can be covered by adding a 1 or can say it will need just 1 more loop to cover them.
Truth Table
ESC_TELEM_MAX_ESCS -> num_idx -> no. of loops by logic-> Actual it should be
4->1->1->1
6->2->2->2
1->1->1->1
5->2->2->2
13->4->4->4

please review it again